Tests/BotTests/Mocks/MockGitHubAPI.swift

```diff
@@ -55,7 +55,7 @@ struct MockGitHubAPI: GitHubAPIProtocol {
         }
     }
 
-    func fetchCommitStatus(for pullRequest: PullRequest) -> SignalProducer<CommitStatus, AnyError> {
+    func fetchCommitStatus(for pullRequest: PullRequest) -> SignalProducer<CommitState, AnyError> {
 
         let index = iteration.modify { iteration -> Int in
```

Comment:
This repeating pattern of
```
        let index = iteration.modify { iteration -> Int in
            iteration = iteration + 1
            return iteration - 1
        }
```
is hard to parse, took me a minute to understand why you'd do a `+1` then a `-1`. And I'm not sure how this `Atomic` API is supposed to be used anyway tbh (like, if the `iteration` closure parameter is `inout` so that you can do `+= 1` to modify it inline, then why does the API also expect to return a value? And if it expects to return the previous value, why didn't they automate that? I'd expect the `Atomic` API to either be `(inout Int) -> Void` or `(Int) -> Int` but not both `inout` and returning a value...

Also the name `iteration` threw me away tbh, wasn't sure what it was supposed to represent

So since it's both a bit confusing and this snippet is repeated multiple times, maybe a renaming and small private func to handle it could help. Like `stubCount` instead of `iteration`, and `func pushNextIndex() -> Int` or something to extract that snippet that both atomically increment the count and return the corresponding index?

Last but not least, not sure if `Atomic` is enough there. I mean, if 2 requests are run concurrently, you might end up incrementing your `iteration` and getting index 0 for the first, then for the second, but then the thread/runloop on which first is ran is paused while request 2 is handled, so you end up returning stub 1 before stub 0, right? At least, not totally sure if that would happen that way but my point is that it might be worth not only wrapping the index increment in an atomic operation, but even the whole access to `stubs`?

Reply:
> Last but not least, not sure if Atomic is enough there. I mean, if 2 requests are run concurrently, you might end up incrementing your iteration and getting index 0 for the first, then for the second, but then the thread/runloop on which first is ran is paused while request 2 is handled, so you end up returning stub 1 before stub 0, right? At least, not totally sure if that would happen that way but my point is that it might be worth not only wrapping the index increment in an atomic operation, but even the whole access to stubs?

I'm missing the problem here because the only thing that we want to protect is the order by which the stubs are requested by, their execution is not controlled in here as we only return `SignalProducer`s which are then started by the consumer so even if we wrap the whole access to `stubs` I don't see how that could prevent or guarantee the correct execution of each producer.

Reply:
Yeah I'm not sure there's really a problem/risk with only protecting the counter. I think I was just confused with that part of the code `+1`/`-1` mainly, and the API of `Atomic` (still not sure why they designed it to be both inout and returning)

Reply:
@AliSoftware it provides extra flexibility having that option, as an example you could try to modify something in your value and either succeed or fail, with this option you can represent that by returning a `Bool` to notify that fact.

Another example is the change that we have done, instead of returning just an index we can return the stub directly and do everything behind the scenes.

```swift
private func nextStub() -> Stubs {
         return stubCount.modify { stubCount -> Stubs in
             let nextStub = stubs[stubCount]
             stubCount = stubCount + 1
             return nextStub
         }
}
```

Modify is all about to change the s

Reply:
Thanks makes way more sense with the new code returning a Stub on how it can be useful indeed 👍